src/seamless_communication/cli/m4t/evaluate/evaluate.py

```diff
@@ -412,27 +439,26 @@ def main(optional_args: Optional[Dict[str, Any]] = None) -> None:
         raise ValueError(
             f"Invalid audio_root_dir: {args.audio_root_dir} for speech input."
         )
-
-    if torch.cuda.is_available():
-        device = torch.device("cuda:0")
-        dtype = torch.float16
-    else:
-        device = torch.device("cpu")
-        dtype = torch.float32
-
-    text_tokenizer = load_unity_text_tokenizer(args.model_name)
+    
+    device = torch.device(args.device)
+    dtype = torch.float16 if device.type == "cuda" else torch.float32
 
     # TODO: Avoid loading the T2U model, vocoder when the output
     # modality is text.
     translator = Translator(
         args.model_name,
         args.vocoder_name,
         device,
-        text_tokenizer=text_tokenizer,
         dtype=dtype,
         input_modality=input_modality,
         output_modality=output_modality,
     )
+    
+    # HACK: Get a quick model load
+    if args.load_checkpoint:
+        saved_model = torch.load(args.load_checkpoint, map_location=device)["model"]
+        saved_model = { k.replace("model.", ""): v for k, v in saved_model.items() }
```

Comment:
why do we strip model. prefix if we already get into model in the previous step?

Reply:
Because the key names also contain `model.`. The dict looks like 
```
{
    "model": {
        "model.speech_encoder...": ...
    }
}
```